token-client/src/test/java/com/sap/cloud/security/client/HttpClientFactoryTest.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * SPDX-FileCopyrightText: 2018-2021 SAP SE or an SAP affiliate company and Cloud Security Client Java contributors
+ * <p>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.cloud.security.client;
+
+import com.sap.cloud.security.config.ClientCredentials;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+
+public class HttpClientFactoryTest {
+
+	@Test
+	public void create() {
+		CloseableHttpClient cut = HttpClientFactory.create(new ClientCredentials("clientId", "secret"));
+		assertNotNull(cut);
+
+		// Assert that custom HttpClientFactory factory has a priority over default
+		// com.sap.cloud.security.client.DefaultHttpClientFactory
+		assertFalse(cut.getClass().getName().contains("InternalHttpClient"));
```

Comment:
this assertion is always true because there is no "InternalHttpClient" at all :-) please check

Reply:
assertTrue(cut.getClass().getName().contains("..."));

Reply:
@nenaraab `InternalHttpClient` is default class resolved when `HttpClients.createDefault()` is used, that's what we have in our 'DefaultHttpClientFactory' implementation used. I could inverse that to check for mocked class instead, but with this check we are also targeting the Default implementation from our side, so I thought it's somewhat more beneficial.

Reply:
ok. But i wasn't able to find "InternalHttpClient" in https://github.com/SAP/cloud-security-xsuaa-integration/blob/main/token-client/src/main/java/com/sap/cloud/security/client/DefaultHttpClientFactory.java?
Maybe I'm wrong, but i would prefer a positive check to make the test bullet proof...